parlai/core/torch_ranker_agent.py

```diff
@@ -104,6 +110,35 @@ def build_model(self):
         raise NotImplementedError(
             'Abstract class: user must implement build_model()')
 
+
+    def get_train_preds(self, scores):
+        if self.opt['candidates'] == 'batch':
+            batchsize = scores.size(0)
+            # get accuracy
+            targets = scores.new_empty(batchsize).long()
+            targets = torch.arange(batchsize, out=targets)
+            nb_ok = (scores.max(dim=1)[1] == targets).float().sum().item()
+            self.metrics['train_accuracy'] += nb_ok
+            # calculate mean rank
+            above_dot_prods = scores - scores.diag().view(-1, 1)
+            rank = (above_dot_prods > 0).float().sum().item()
+            self.metrics['rank'] += rank
+            return Output()
```

Comment:
this section looks pretty cheap to compute. if that's true (@emilydinan can you verify?), then I think we should leave it on by default always (no need to gate a cheap metric behind a flag, better to have more metrics available without having to rerun your script with a different flag to see them)

Reply:
verified, will update